.github/workflows/main.yml

```diff
@@ -6,52 +6,74 @@ on:
       - main
 
 jobs:
+  validation:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: gradle/wrapper-validation-action@v1
+
   checks:
     runs-on: macos-latest
     steps:
-
       - name: Project Checkout
-        uses: actions/checkout@v2
-
+        uses: actions/checkout@v3
       - name: Setup JDK 1.8
-        uses: actions/setup-java@v2
+        uses: actions/setup-java@v3
         with:
           java-version: 8
           distribution: 'zulu'
-
       - name: Build project
-        run: ./gradlew build --stacktrace
-
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: build --stacktrace
```

Comment:
https://github.com/gradle/gradle-build-action

Ref https://github.com/detekt/detekt/pull/3723, it's really so fast to run jobs, and many other advantages.